packages/core/src/util/runtime.ts

```diff
@@ -177,3 +180,51 @@ export const isEnabled = (
 
   return true;
 };
+
+export const isInherentlyEnabled = (
+  state: JsonFormsState,
+  ownProps: any,
+  uischema: UISchemaElement,
+  schema: JsonSchema & { readonly?: boolean; readOnly?: boolean },
+  rootData: any,
+  config: any
+) => {
+  if (state?.jsonforms?.readonly) {
+    return false;
+  }
+  if (
+    uischema &&
+    hasEnableRule(uischema) &&
+    !isEnabled(uischema, rootData, ownProps?.path, getAjv(state))
+  ) {
+    return false;
+  }
+  if (
+    typeof uischema?.options?.readonly === 'boolean' &&
+    uischema.options.readonly
+  ) {
+    return false;
+  }
+  if (
+    typeof uischema?.options?.readOnly === 'boolean' &&
+    uischema.options.readOnly
+  ) {
+    return false;
+  }
+  if (typeof config?.readonly === 'boolean' && config.readonly) {
+    return false;
+  }
+  if (typeof config?.readOnly === 'boolean' && config.readOnly) {
+    return false;
+  }
+  if (typeof schema?.readonly === 'boolean' && schema.readonly) {
+    return false;
+  }
+  if (typeof schema?.readOnly === 'boolean' && schema.readOnly) {
+    return false;
+  }
+  if (typeof ownProps?.enabled === 'boolean') {
+    return ownProps.enabled;
+  }
+  return true;
+};
```

Comment:
This is still performing an `OR` operation ;) I wanted it to look like this:
```suggestion
  if (state?.jsonforms?.readonly) {
    return false;
  }
  if (
    uischema &&
    hasEnableRule(uischema)
  ) {
    return isEnabled(uischema, rootData, ownProps?.path, getAjv(state));
  }
  if (
    typeof uischema?.options?.readonly === 'boolean'
  ) {
    return !uischema.options.readonly;
  }
  if (
    typeof uischema?.options?.readOnly === 'boolean'
  ) {
    return !uischema.options.readOnly;
  }
  if (typeof config?.readonly === 'boolean') {
    return !config.readonly;
  }
  if (typeof config?.readOnly === 'boolean') {
    return !config.readOnly;
  }
  if (typeof schema?.readOnly === 'boolean') {
    return !schema.readOnly;
  }
  if (typeof ownProps?.enabled === 'boolean') {
    return ownProps.enabled;
  }
  return true;
};
```

Reply:
Hey, I've done this at first but I noticed it's killing the whole override mentality you built. But if you need that sure thing, it will my pleasure to do it. On it.

Reply:
The question for me is where to place the rule evaluation. Whether it makes more sense at the top, like it is right now, or whether local uischema.options can override them, i.e. they should be placed to before the `schema.readOnly` check.

Reply:
That's the thing I don't know. It feels more natural to me to keep the original behaviour as is except if there is an explicit "readonly" on the schema of the field or respective uischema. What do you think?

Reply:
By the way no luck on running the test (vue part is failing to run) due to limitations of M1 I have to use node 15 :(

Reply:
I updated the algorithm per your suggestion